components/tag-list/tag-list-item-mixin.js

```diff
@@ -169,12 +175,12 @@ export const TagListItemMixin = superclass => class extends LocalizeCoreElement(
 	_handleClearItem() {
 		if (!this.clearable) return;
 
-		announce(this.localize('components.tag-list.cleared-item', { value: this.text }));
+		announce(this.localize('components.tag-list.cleared-item', { value: this._displayText }));
 
 		/** Dispatched when a user selects to delete an individual tag list item. The consumer must handle the actual element deletion and focus behaviour if there are no remaining list items. */
 		this.dispatchEvent(new CustomEvent(
 			'd2l-tag-list-item-clear',
-			{ bubbles: true, composed: true, detail: { value: this.text } }
+			{ bubbles: true, composed: true, detail: { key: this.key } }
```

Comment:
I don't think there's much value in this.
```suggestion
			{ bubbles: true, composed: true }
```

Reply:
Yeah I agree, but do you think it'd help people "do the right thing"? Encourages using the event detail to get the data from the data array, rather than hooking up an anonymous function to each item passing in data like a few places are doing?

Reply:
Yeah, I guess it could, and it doesn't hurt any to leave it here.

Reply:
I'm not sure it's going to prevent that. If they're not inclined to use `e.target.key` then I suspect they won't use `e.detail.key` either. I usually opt for not duplicating the data that just as easily accessed, but as you said, it doesn't really hurt.